### ISSUE TITLE
Unclear installation instructions

### ISSUE THREAD

**Issue by uuf6429** [Kotlin/kotlinx.serialization#2509]
I've spent a ridiculous amount of time trying to set this up, with no success whatsoever.

Let's look at the instructions...
<img width="244" alt="image" src="https://github.com/Kotlin/kotlinx.serialization/assets/230049/400851d2-8ad2-4961-8205-b746dd86e38e">

To me that means that we either install it in gradle or in android, or in multiplatform...
Otherwise the headings do not make any sense. Maybe the headings should have been so:
- Gradle
   1. Setting up the serialization plugin
   2. Dependency on the json library
      - Android
      - Multiplatform
- Maven

??

Next, why is this library depending on a plugin? When do we need to install this plugin...is it optional or mandatory?

Why is "android" separate from "multiplatform"? Isn't android just another platform?

Looking at the instructions themselves... I have 3 `*.gradle.kts` files.. where should the setup settings go?

Sidenote: should end users of this library really care about the proguard stuff mentioned under "android" in the setup?

----

So in short (since I can't get android to compile and IDE completion fails randomly), what are the real instructions to use this in a KMP project?

**Comment by uuf6429** [comment]
After some trial and error, here's what seems to make it work (at least on android):

In `gradle/libs.versions.toml`, I have the following:
```toml
# under the [versions] section
jetbrains-kotlinxSerializationJson = "1.0.0"
jetbrains-kotlinxSerializationRuntime = "0.14.0"
kotlinxSerializationJson = "1.4.1"

# under the [libraries] section
jetbrains-kotlinx-serialization-runtime = { module = "org.jetbrains.kotlinx:kotlinx-serialization-runtime", version.ref = "jetbrains-kotlinxSerializationRuntime" }
kotlinx-serialization-json = { module = "org.jetbrains.kotlinx:kotlinx-serialization-json", version.ref = "kotlinxSerializationJson" }

# under the [plugins] section
kotlinSerialization = { id = "org.jetbrains.kotlin.plugin.serialization", version.ref = "kotlin" }
```

And in `composeApp/build.gradle.kts`:
```kts
plugins {
// ...
    alias(libs.plugins.kotlinSerialization)     // <----
}

kotlin {
// ...

    sourceSets {
        val androidMain by getting {
            dependencies {
// ...
            }
        }
        val commonMain by getting {
            dependencies {
// ...
                implementation(libs.kotlinx.serialization.json)     // <----
            }
        }
    }
}
```



Most of the problem here is the (usual) catastrophic mess surrounding gradle. It's beyond me why a great product (kotlin) has to depend on it. It would be best to expect/assume that the end users are clueless about gradle and therefore be extra careful documenting the installation instructions.

**Comment by sandwwraith** [comment]
Hello, I believe it is easier for me to answer your questions one-by-one:

> To me that means that we either install it in gradle or in android, or in multiplatform

Given that Gradle is a default build system for Android and KMP projects, it is a strange conclusion. How can you install it in Android but not in Gradle?

> Next, why is this library depending on a plugin?

The plugin is responsible for generating serialization code, as we do not use reflection. You can read about why a plugin is necessary and about the general architecture of kotlin.serialization here: https://github.com/Kotlin/KEEP/blob/master/proposals/extensions/serialization.md

> When do we need to install this plugin...is it optional or mandatory?

It is emphasized in the instructions: `To set up kotlinx.serialization, you have to do two things:` under `Gradle` section and `After setting up the plugin, you have to add a dependency on the serialization library.` under `2) Dependency on the JSON library`.

> Why is "android" separate from "multiplatform"? Isn't android just another platform?

Because this library can be used as a regular Android-alone project as well as a part of KMP project.

> Sidenote: should end users of this library really care about the proguard stuff mentioned under "android" in the setup?

Yes, because the proguard/r8 is an important piece of technology used mainly in Android development.

> jetbrains-kotlinxSerializationJson = "1.0.0"

It is an outdated version. Check out releases on github: we have 1.6.1 now. The same version is also demonstrated in the instructions directly.

> jetbrains-kotlinxSerializationRuntime = "0.14.0"

It is an obsolete version, and it is not recommended to use. You do not depend on it anyway.

>  It would be best to expect/assume that the end users are clueless about gradle and therefore be extra careful documenting the installation instructions.

I strongly disagree with you here. Knowledge of the build system (to some reasonable extent) is just as important as knowledge of the language and other technologies (e.g. Proguard or Android SDK). So being 'clueless' is just lacking qualification for the job.

I hope I answered all of your questions. If you have further ones, you can re-open this issue.
